### PR TITLE
feat(Requirements/Boost): added boost support

### PR DESCRIPTION
**Pull request by Winfidonarleyan**
## Related Issue
- https://github.com/azerothcore/azerothcore-wotlk/pull/5676

### CODE REVIEW

File: docs/windows-requirements.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
 
 | |
 | :- |
+| Boost ≥ 1.70 (latest stable recommended) |
```
Comment (Kitzunu) [inline docs/windows-requirements.md:10]: Does it work with 1.76

File: docs/windows-requirements.md [hunk 2]
```diff
@@ -72,7 +73,17 @@
        
     1. *Note #2: While installing OpenSSL, choose The OpenSSL binaries (/bin) directory (NOT "The Windows system directory")*
        *when given the choice on where to copy the OpenSSL DLLs. These DLLs will need to be located easily for [Core Installation](windows-core-installation).*
+  
+1. Boost.
```
Comment (Kitzunu) [inline docs/windows-requirements.md:77]:
```
[Boost](https://www.boost.org/)
```

File: docs/windows-requirements.md [hunk 3]
```diff
@@ -72,7 +73,17 @@
...
 
+    1. Download the prebuilt Windows Binary for Visual Studio 2019
+
+    3. `1.70.0` is the minimum version required for Visual Studio 2019, but Version `1.74.0`is recommended
+
+    5. 64bit: https://sourceforge.net/projects/boost/files/boost-binaries/1.74.0/boost_1_74_0-msvc-14.2-64.exe/download
+
+    7. 32bit: https://sourceforge.net/projects/boost/files/boost-binaries/1.74.0/boost_1_74_0-msvc-14.2-32.exe/download
```
Comment (Kitzunu) [inline docs/windows-requirements.md:85]: Make all "1."

File: docs/windows-requirements.md (hunk 1 above)
Comment (Winfidonarleyan) [reply docs/windows-requirements.md:10]: now `Cmake` don't support 1.76 boost

Comment (Winfidonarleyan) [reply docs/windows-requirements.md:10]: if apply commit https://gitlab.kitware.com/cmake/cmake/-/commit/79be37b94e7353ec056e6343ba293ce1b60627d4 - no problems

Comment (Kitzunu) [reply docs/windows-requirements.md:10]: Maybe better to not say the latest stable then but instead recommend 1.74?

Comment (Winfidonarleyan) [reply docs/windows-requirements.md:10]: Hm, maybe better

File: docs/windows-requirements.md [hunk 4]
```diff
@@ -7,6 +7,7 @@
 
 | |
 | :- |
+| Boost ≥ 1.70 |
```
Comment (Kitzunu) [inline docs/windows-requirements.md:10]:
```suggestion
| Boost ≥ 1.70 - 1.74 |
```

**Review by Kitzunu — APPROVED** [review]

**Pull request merged**